app/src/org/commcare/activities/connect/ConnectIdPhoneVerificationActivity.java

```
@@ -74,6 +90,56 @@ protected void onCreate(Bundle savedInstanceState) {
         startHandler();
     }
 
+    @Override
+    protected void onStart() {
+        super.onStart();
+        registerBrodcastReciever();
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if(requestCode == REQ_USER_CONSENT){
+            if((resultCode== RESULT_OK) && data != null){
+                String message= data.getStringExtra(SmsRetriever.EXTRA_SMS_MESSAGE);
+                getOtpFromMessage(message);
+
+            }
+        }
+    }
+
+    private void getOtpFromMessage(String message) {
+
+        Pattern otpPattern = Pattern.compile("(|^)\\d{6}");
+        Matcher matcher = otpPattern.matcher(message);
+        if (matcher.find()){
+
+            uiController.setCode(matcher.group(0));
+        }
+
+
+    }
+
+    public void registerBrodcastReciever(){
+        smsBroadcastReceiver = new MySMSBroadcastReceiver();
+
+        smsBroadcastReceiver.smsListener = new SMSListener() {
+            @Override
+            public void onSuccess(Intent intent) {
+                    startActivityForResult(intent,REQ_USER_CONSENT);
+            }
+
+            @Override
+            public void onError(String message) {
+                showToastMessage(message);
```

We don't need to tell the user if auto-retrieval fails, just fail silently